### DISCUSSION TITLE
icon problem when publish ipa file

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2429] [Q&A]
You are all welcome
I have gone through many problems and no pain why all this is happening.
This problem occurred when I uploaded the application to the store through the program (Transporter)
Knowing that I put the icons inside a folder named (icon) and I put all the icons for the application and I do not know the solution to this problem and I use (adobe animate ) and use harman 50.1.1.2 ios and I have a program () on the device with the latest version
![Screen Shot 2023-01-16 at 4 43 22 PM](https://user-images.githubusercontent.com/51138931/212707911-e4a9855d-d1e3-4b54-b6e7-c59edee2b4f6.png)

**Comment by FliplineStudios** [comment]
Could you share how you've set up your icons for the app, to make sure you've included all of those sizes?  In general, I believe this is what needs to be included:

- Define each icon size in the `<icon>` section of the application XML (for example, `<image120x120>your120IconHere.png</image120x120>`)
- If your icon PNGs are in a folder, make sure that folder is packaged along with the app (if you're using Animate, have that folder added to the "Included Files" section)
- Create an Assets.car file using Xcode or other tools, which contains all of the required icon sizes
- Make sure the Assets.car file is packaged along with the app as well in "Included Files"

**Reply by avcr2019** [reply]
> Could you share how you've set up your icons for the app, to make sure you've included all of those sizes? In general, I believe this is what needs to be included:
> 
> * Define each icon size in the `<icon>` section of the application XML (for example, `<image120x120>your120IconHere.png</image120x120>`)
> * If your icon PNGs are in a folder, make sure that folder is packaged along with the app (if you're using Animate, have that folder added to the "Included Files" section)
> * Create an Assets.car file using Xcode or other tools, which contains all of the required icon sizes
> * Make sure the Assets.car file is packaged along with the app as well in "Included Files"

thank you bro . can you make video please i cant make it .

**Comment by marchbold** [comment]
You likely need to generate an "asset catalogue" (`Assets.car`) for your app. Have a look here at methods to create it:

https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car

**Reply by avcr2019** [reply]
> You likely need to generate an "asset catalogue" (`Assets.car`) for your app. Have a look here at methods to create it:
> 
> https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car

thank you my friend
can you make video for this i cant make it perfect  please >

**Reply by avcr2019** [reply]
> You likely need to generate an "asset catalogue" (`Assets.car`) for your app. Have a look here at methods to create it:
> 
> https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car
please help me . There was a big problem during the experiment, but with another file that I found after your file. I will send it to you. The problem is that during the experiment and running the executable file, the file erased all the contents of the folder, including my work files, and I do not know where I went. This work project I have been working on for about 6 months, so how do I restore it please. I will cry because of this
![Screen Shot 2023-01-17 at 11 58 53 AM](https://user-images.githubusercontent.com/51138931/212868487-6fd3017a-0e47-4fb4-a19a-21d38a6fd5db.png)


I have attached the file here, I want you to look at it and the executable file, and tell me what to do to restore the files that were deleted
[assets-car-build.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10433723/assets-car-build.zip)

**Comment by avcr2019** [comment]
please any one here can help me to restore my file of my project ??? i used this file and it delet all my file 
[assets-car-build.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10435089/assets-car-build.zip)

**Reply by marchbold** [reply]
How did you run the script? 

It should be placed in an new directory with your launch and icon pngs and  then executed in terminal. 

It should only cleanup the tmp files it creates.

**Reply by avcr2019** [reply]
> How did you run the script?
> 
> It should be placed in an new directory with your launch and icon pngs and then executed in terminal.
> 
> It should only cleanup the tmp files it creates.

I don't know how it works. I read that you must run the file and click on it, and it will create a file named (()). This is what I did and what happened was not taken into account. I lost my entire project and I am very sad and I don't know what to say my dear friend. I installed a new system from scratch, and unfortunately I will restart my project, and months and months of effort have been wasted. I ask you for a proper way to implement the icons and extract the file safely, and I ask you to make a simple video and attach it here, if you please, my dear friend, because I am afraid to try again. I hope that you will record a simple video or steps with pictures in detail, and I will do it because I will need to make these icons after making the application. Thank you with all my heart .

**Reply by marchbold** [reply]
Never click on a script, this will generally have unexpected results! It says in the instructions to open a terminal and run the script in a directory with your icon and launch image. 

Did you have your code in version control ? Perhaps try restoring?

**Reply by avcr2019** [reply]
> Never click on a script, this will generally have unexpected results! It says in the instructions to open a terminal and run the script in a directory with your icon and launch image.
> 
> Did you have your code in version control ? Perhaps try restoring?

Unfortunately, I did not understand the explanation directly. Unfortunately, I pressed the executable file directly and deleted all the files on the device, even the external hard drive was connected to the device and also deleted its files. I was shocked and still am more shocked by the loss of my files. I wish you, my friend, to make a video explanation or a detailed explanation with pictures so that I can extract the icons file properly and correctly, because it seems wrong on my part to use or apply the explanation. As for the files that you lost, you will not come back again, unfortunately, and I tried a lot, but to no avail. Thank you for the help and for the reply. I wish you a happy day and do not forget me to explain the method in detail

**Reply by ajwfrost** [reply]
From my reading of that script it should only have deleted everything under `/build` .. not sure how you can actually delete "everything" on the computer given that a lot of the essential stuff would be protected under root ownership, but user-specific files yes that could be a problem.

But -> when you do actually "rm" a file, it only unlinks the pointer to the file itself, the data would stay there. So it should be possible to take this to a computer specialist and get them to restore the files, you may need to add back in the filenames or similar, but the bulk of the data would be available I hope.

**Reply by marchbold** [reply]
Also, where did you get that script as it's not the one I linked you too? It's a very old script I wrote many years ago and really shouldn't be used.

**Reply by avcr2019** [reply]
> From my reading of that script it should only have deleted everything under `/build` .. not sure how you can actually delete "everything" on the computer given that a lot of the essential stuff would be protected under root ownership, but user-specific files yes that could be a problem.
> 
> But -> when you do actually "rm" a file, it only unlinks the pointer to the file itself, the data would stay there. So it should be possible to take this to a computer specialist and get them to restore the files, you may need to add back in the filenames or similar, but the bulk of the data would be available I hope.

Unfortunately, I did not know how to use the file, and I did not know that the file was old and I used it in the wrong way. Unfortunately, it completely erased all the files of the device, and I could not recover the files. Thank you for watching, my dear friend. I wish you a happy day

**Reply by avcr2019** [reply]
> Also, where did you get that script as it's not the one I linked you too? It's a very old script I wrote many years ago and really shouldn't be used.


Actually, I used the old one by mistake. I apologize for that, my friend. I hope that you will explain the new file because I want to publish a small part as an experiment on Apple. Until now, I am afraid to try any executable file on the device, especially since I started working on the project again. Can you make a simple illustrated or written explanation in detail so that I can do the icons correctly. I wish you a happy day . Eagerly waiting for your reply

**Reply by marchbold** [reply]
Okay, if you could let me know where you got it from though I'll see if I can get it taken down.

There is a tutorial here on how to use the script: 

https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car

and more details in the repository

https://github.com/distriqt/AIR-ImageScripts

Let me know if you need more information.

**Reply by avcr2019** [reply]
> Okay, if you could let me know where you got it from though I'll see if I can get it taken down.
> 
> There is a tutorial here on how to use the script:
> 
> https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car
> 
> and more details in the repository
> 
> https://github.com/distriqt/AIR-ImageScripts
> 
> Let me know if you need more information.

Thank you for your reply, my dear friend.
For the file that I downloaded and caused me problems. I do not remember where the download link is because all data on the device was erased because of the file. But here on the site one of the members almost put it to one of the members when he had the same problem.
For the explanation links that you sent. I thank you for these links, but I am afraid of the first method, which contains codes, so I asked you to record a video, even if it was just a few seconds, in order for me to conduct the experiment successfully. Because I tried the second method on (X code) and did not find the option (single view app), so I chose (app), and when I reached here (Build the project ( Product -> Build).
Right-click on your ‘.app’ -> Show in finder.
Right click on your ‘.app’ -> Show package contents.
Now copy Assets.car and package with AIR application.), I could not continue. As for the third method, it works successfully, but I prefer to try the first or second method, especially since there is better control. I wish you to record a few seconds for the first or second method, as you like, because I am nervous and afraid of the first method. Because I feel I will make a mistake again and it will erase all my files. Thank you and waiting for your reply my friend

**Reply by avcr2019** [reply]
> Okay, if you could let me know where you got it from though I'll see if I can get it taken down.
> 
> There is a tutorial here on how to use the script:
> 
> https://docs.airnativeextensions.com/docs/tutorials/ios-icons-assets-car
> 
> and more details in the repository
> 
> https://github.com/distriqt/AIR-ImageScripts
> 
> Let me know if you need more information.

Excuse me, my friend
Is this(storyboard) of great use and should I add it in the application or is it optional

**Reply by marchbold** [reply]
If you have a license for AIR you should add a storyboard. It sets up the launch screen like the old Default.png did in older iOS versions. 

There's not a lot to this so not sure a video would help much, open up a terminal and use the following commands:

```
git clone git@github.com:distriqt/AIR-ImageScripts.git
cd AIR-ImageScripts
```

In this directory update the example `icon.png` and `launch.png` for your app, then run:

```
./generate.sh
```

This will generate an `out` directory with the `Assets.car` and `LaunchScreen.storyboardc` and `icons` directory.

**Reply by avcr2019** [reply]
> If you have a license for AIR you should add a storyboard. It sets up the launch screen like the old Default.png did in older iOS versions.
> 
> There's not a lot to this so not sure a video would help much, open up a terminal and use the following commands:
> 
> ```
> git clone git@github.com:distriqt/AIR-ImageScripts.git
> cd AIR-ImageScripts
> ```
> 
> In this directory update the example `icon.png` and `launch.png` for your app, then run:
> 
> ```
> ./generate.sh
> ```
> 
> This will generate an `out` directory with the `Assets.car` and `LaunchScreen.storyboardc` and `icons` directory.

Thank you dear friend . I wish you a happy day
